### ISSUE TITLE
infer code

### ISSUE THREAD

**Issue by moonhhh-mo** [BUTSpeechFIT/DiCoW#8]
How is it converted into inference code using Hugging Face and Transformers? I've already fine-tuned the entire Dicow model.😊 Directly replacing your original model weights doesn't seem to work.

**Comment by Lakoc** [comment]
Hello, which model did you fine-tune? Could you provide any additional details? Without them, it’s a bit hard to determine where the issue lies. In about a week, we will be releasing a new extended version of the TS-ASR-Whisper repo, which will include the following export script.

```python
from src.models.dicow.modeling_dicow import DiCoWForConditionalGeneration
from src.models.dicow.config import DiCoWConfig
from huggingface_hub import HfApi
import os
from transformers import AutoTokenizer, AutoFeatureExtractor

if __name__ == '__main__':
    CHECKPOINT_PATH = "........."
    ORG = "BUT-FIT"
    MODEL_NAME = "SE_DiCoW"
    NEW_MODEL_ID = f"{ORG}/{MODEL_NAME}"
    DiCoWConfig.register_for_auto_class()
    DiCoWForConditionalGeneration.register_for_auto_class("AutoModelForSpeechSeq2Seq")
    api = HfApi()

    dicow_pretrained = ("CHECKPOINT_PATH")
    dicow_config = DiCoWConfig(**dicow_pretrained.config.to_dict())
    dicow_model = DiCoWForConditionalGeneration(dicow_config)
    dicow_model.load_state_dict(dicow_pretrained.state_dict())
    fe = AutoFeatureExtractor.from_pretrained("openai/whisper-large-v3-turbo")
    tokenizer = AutoTokenizer.from_pretrained("openai/whisper-large-v3-turbo")

    dicow_model.push_to_hub(NEW_MODEL_ID)
    fe.push_to_hub(NEW_MODEL_ID)
    tokenizer.push_to_hub(NEW_MODEL_ID)
    api.upload_file(
        path_or_fileobj=f"{os.environ['SRC_ROOT']}/export_sources/readmes/{MODEL_NAME}.md",
        path_in_repo="README.md",
        repo_id=NEW_MODEL_ID,
    )
    api.upload_file(
        path_or_fileobj=f"{os.environ['SRC_ROOT']}/export_sources/generation_config.json",
        path_in_repo="generation_config.json",
        repo_id=NEW_MODEL_ID,
    )
```

**Comment by moonhhh-mo** [comment]
Thanks, I carried out the pretraining with CTC and fine‑tuned the entire Whisper‑Large‑V3 model. Now I’d like to upload it to my own Hugging Face account and use Transformers for inference, just like in the code you provided.

**Comment by Lakoc** [comment]
Great — you just need to make sure all local imports are relative and come only from the folder where the main model class is defined.

**Comment by moonhhh-mo** [comment]
Thank you, I’ll give it a try. If I run into any problems, I’ll consult you again.😁

**Comment by moonhhh-mo** [comment]
I’m facing a new problem: the results don’t distinguish between speakers — the outputs for both speakers are identical. In other words, during inference the **model generates the same tokens for different speakers**, whereas with your model the outputs are different. To investigate, I checked the model’s outputs during evaluation, and they were normal. At inference time, I also checked inputs such as the "stno_mask", and they are the same as when using your model. What could be the reason for this?

**Comment by moonhhh-mo** [comment]
For example, the result is:

Spk-1:
<|0.00|>11111111<|0.04|>
<|0.10|>333333<|1.23|>
.....

Spk-2:
<|0.00|>11111111<|0.04|>
<|0.10|>333333<|1.23|>
.....

**Comment by Lakoc** [comment]
Again, without knowing your fine-tuning setup and the data you used, it's very hard to tell. I would check whether your fine-tuning dataset has the correct labels.

**Comment by Lakoc** [comment]
I’m closing this issue for now since there hasn’t been any recent activity. Don’t hesitate to reopen it if you have more questions or need further assistance!
